editor/editor_file_system.cpp

```diff
@@ -2072,15 +2072,17 @@ void EditorFileSystem::update_files(const Vector<String> &p_script_paths) {
 	}
 
 	if (updated) {
-		_process_update_pending();
 		if (update_files_icon_cache) {
 			_update_files_icon_path();
 		} else {
 			for (EditorFileSystemDirectory::FileInfo *fi : files_to_update_icon_path) {
 				_update_file_icon_path(fi);
 			}
 		}
-		call_deferred(SNAME("emit_signal"), "filesystem_changed"); //update later
+		if (!is_scanning()) {
+			_process_update_pending();
+			call_deferred(SNAME("emit_signal"), "filesystem_changed"); //update later
```

Comment:
```suggestion
			call_deferred(SNAME("emit_signal"), "filesystem_changed"); // Update later.
```

Reply:
Forgot about this before merging. That will be for the next PR in that code :D